### PR TITLE
WIP: Update Spark Operator to support ocp 4.9

### PR DESCRIPTION
**Pull request by LaVLaS**
Replaces #506 and adds changes to resolve issues with new OCP 4.9 requirements

Instructions to test:
* Deploy ODH with JupterHub and Spark operator enabled
* Confirm JupyterHub and Spark operator pods are running successfully
* Login to JupyterHub and Spawn a spark notebook.
* Confirm that the notebook spawns successfully and a user spark cluster pods are deployed

### CODE REVIEW

File: radanalyticsio/spark/cluster/base/role.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  resources: ["routes"]
+  verbs: ["create", "delete", "deletecollection", "get", "list", "patch", "update", "watch"]
+- apiGroups: [""]
+  resources: ["persistentvolumeclaims"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["create", "delete", "deletecollection", "get", "list", "patch", "update", "watch"]
+- apiGroups: ["radanalytics.io/v1"]
+  resources: ["SparkCluster","SparkApplication","SparkHistoryServer"]
```
Comment (HumairAK) [inline radanalyticsio/spark/cluster/base/role.yaml:22]: nit: add spaces between commas

**Comment by HumairAK** [comment]
Tried this in operate-first environment: Encountered the following error in spark-operator pod: 

```
2022-02-15 20:45:12 INFO  SDKEntrypoint:108 - [33mOpenShift[0m environment detected.
2022-02-15 20:45:12 INFO  AbstractOperator:255 - Starting 'SparkHistoryServer' operator for namespace *
2022-02-15 20:45:16 INFO  CrdDeployer:51 - CustomResourceDefinition for SparkHistoryServer has been found in the K8s, so we are skipping the creation.
2022-02-15 20:45:16 INFO  SDKEntrypoint:199 - full reconciliation for 'SparkHistoryServer' operator scheduled (periodically each 180 seconds)
2022-02-15 20:45:16 INFO  SDKEntrypoint:200 - the first full reconciliation for 'SparkHistoryServer' operator is happening in 2 seconds
2022-02-15 20:45:16 INFO  AbstractOperator:255 - Starting 'SparkCluster' operator for namespace *
2022-02-15 20:45:16 ERROR AbstractOperator:280 - 'SparkHistoryServer' operator startup failed for namespace *
java.lang.NullPointerException
	at java.base/java.net.URI$Parser.parse(URI.java:3104)
	at java.base/java.net.URI.<init>(URI.java:600)
	at io.fabric8.kubernetes.client.utils.URLUtils.join(URLUtils.java:50)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.getRootUrl(OperationSupport.java:152)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.getNamespacedUrl(OperationSupport.java:159)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.getNamespacedUrl(OperationSupport.java:170)
	at io.fabric8.kubernetes.client.dsl.internal.BaseOperationRequestBuilder.<init>(BaseOperationRequestBuilder.java:39)
	at io.fabric8.kubernetes.client.dsl.internal.AbstractWatchManager.<init>(AbstractWatchManager.java:76)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.<init>(WatchConnectionManager.java:63)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.<init>(WatchConnectionManager.java:70)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:577)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:554)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:91)
	at io.radanalytics.operator.common.AbstractWatcher.lambda$createCustomResourceWatch$3(AbstractWatcher.java:127)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
2022-02-15 20:45:16 ERROR SDKEntrypoint:182 - 'SparkHistoryServer' operator in namespace * failed to start

```

**Comment by LaVLaS** [comment]
A status update for anyone following this PR.  The Radanalytics Spark Operator code will require some non-trivial refactoring to support dependency version updates in response to k8s 1.22 API updates.   Since Radanalytics is no longer being maintained, we are discussing the best path forward with functionality provided by the spark operator

**Comment by LaVLaS** [comment]
Since the Radanalytics Spark Operator is no longer supported, I am officially closing this PR.  There is a [Spark on OpenShift](https://github.com/opendatahub-io-contrib/spark-on-openshift) project that is available under the [ODH Contrib](https://github.com/opendatahub-io-contrib) organization

**Comment by diwilli** [comment]
Sorry to be that person to comment on a closed ticket, is there a documentation up required to point people in the right direction, rather than relying on them to stumble over this PR?

**Pull request closed without merging**